script/tool/lib/src/pubspec_check_command.dart

```diff
@@ -426,4 +490,51 @@ class PubspecCheckCommand extends PackageLoopingCommand {
     }
     return result ?? Version.none;
   }
+
+  // Validates the dependencies for a package, returning an error string if
+  // there are any that aren't allowed.
+  String? _checkDependencies(Pubspec pubspec) {
+    final Set<String> badDependencies = <String>{};
+    for (final Map<String, Dependency> dependencies
+        in <Map<String, Dependency>>[
+      pubspec.dependencies,
+      pubspec.devDependencies
+    ]) {
+      for (final MapEntry<String, Dependency> entry in dependencies.entries) {
+        final String name = entry.key;
+        if (!_shouldAllowDependency(name, entry.value)) {
+          badDependencies.add(name);
+        }
+      }
```

Comment:
Alternatively to avoid the `MapEntry`
```suggestion
      dependencies.forEach((String name, Dependency dependency) {
        if (!_shouldAllowDependency(name, dependency)) {
          badDependencies.add(name);
        }
      });
```

Reply:
Nice; I stopped using forEach because of the lint about it, and didn't realize it didn't apply to the map version.